docs/assembly.rst

```diff
@@ -391,12 +394,35 @@ will have a wrong impression about the stack height at label ``two``:
 .. code::
 
     {
+        let x := 8
         jump(two)
         one:
-            // Here the stack height is 1 (because we pushed 7),
-            // but the assembler thinks it is 0 because it reads
+            // Here the stack height is 2 (because we pushed x and 7),
+            // but the assembler thinks it is 1 because it reads
             // from top to bottom.
-            // Accessing stack variables here will lead to errors.
+            // Accessing the stack variable x here will lead to errors.
+            x := 9
```

Comment:
Why are assignments needed in the assembly language?  This is a source of complication.  Since assignments exist, variables do not just denote values, but locations on the stack.  Since variables denote locations on the stack, the stack height adjustments are problematic.  So, instead of introducing labels with height adjustments, you can remove assignments.  I guess we seek simplicity, and less features contribute to simplicity.